src/test/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbeTest.java

```diff
@@ -0,0 +1,91 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.nio.file.Files;
+import java.util.Map;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class LastCommitDateProbeTest {
+    @Test
+    public void shouldKeepScmAsKey() {
+        assertThat(new LastCommitDateProbe().key()).isEqualTo("last-commit-date");
+    }
+
+    @Test
+    public void shouldNotRequireRelease() {
+        assertThat(new LastCommitDateProbe().requiresRelease()).isFalse();
+    }
+
+    @Test
+    public void shouldHaveValidDescription() {
+        assertThat(new LastCommitDateProbe().getDescription()).isNotEqualTo(new Probe() {
+            @Override
+            protected ProbeResult doApply(Plugin plugin) {
+                return null;
+            }
+
+            @Override
+            public String key() {
+                return null;
+            }
+        }.getDescription());
+    }
+
+    @Test
+    public void shouldBeExecutedAfterSCMLinkValidation() {
+        assertThat(SCMLinkValidationProbe.ORDER).isLessThan(LastCommitDateProbe.ORDER);
+    }
+
+    @Test
+    public void shouldReturnSuccessStatusOnValidSCM() {
+        final Plugin plugin = mock(Plugin.class);
+        final LastCommitDateProbe probe = new LastCommitDateProbe();
+        assertNull(probe.tempDirectory);
+
+        when(plugin.getDetails()).thenReturn(Map.of(SCMLinkValidationProbe.KEY, ProbeResult.success("scm", "The plugin SCM link is valid")));
+        when(plugin.getScm()).thenReturn("https://github.com/jenkinsci/parameterized-trigger-plugin.git");
+        when(plugin.getName()).thenReturn("parameterized-trigger");
+        final ProbeResult r = probe.apply(plugin);
+
+        assertNotNull(probe.tempDirectory);
+        assertFalse(Files.exists(probe.tempDirectory));
```

Comment:
Just out of curiosity, do we really need an integration test to check if the plugin folder gets created and then gets deleted in the end or not?

Because here I'm asserting at the beginning of the tempDirectory for the current probe is null (Line 57), and it's non-null after we have run `probe.apply()` (Line 64) signifying that a tempDirectory has been created. And then I assert False to the existence of any folder at the `probe.tempDirectory` path (Line 65), signifying the folder has been deleted.

This test fails if I comment out the code which deletes the plugin directory.

Reply:
`tempDirectory` should definitively **not** be public. 
This is the case in 99% of the situations: class field should **not** be public.